sled-hardware/types/src/lib.rs

```diff
@@ -4,10 +4,48 @@
 
 use schemars::JsonSchema;
 use serde::{Deserialize, Serialize};
+use std::ops::RangeInclusive;
 
 pub mod underlay;
 
-/// Describes properties that should uniquely identify a Gimlet.
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum OxideSled {
+    Gimlet,
+    Cosmo,
+}
+
+impl OxideSled {
+    pub fn try_from_root_node_name(root_node_name: &str) -> Option<Self> {
+        const GIMLET_ROOT_NODE_NAME: &str = "Oxide,Gimlet";
+        const COSMO_ROOT_NODE_NAME: &str = "Oxide,Cosmo";
+        match root_node_name {
+            GIMLET_ROOT_NODE_NAME => Some(Self::Gimlet),
+            COSMO_ROOT_NODE_NAME => Some(Self::Cosmo),
+            _ => None,
+        }
+    }
+
+    pub fn m2_disk_slots(&self) -> RangeInclusive<i64> {
+        match self {
+            Self::Gimlet | Self::Cosmo => 0x11..=0x12,
+        }
+    }
+
+    pub fn u2_disk_slots(&self) -> RangeInclusive<i64> {
+        match self {
+            Self::Gimlet => 0x00..=0x09,
+            Self::Cosmo => 0x20..=0x29,
+        }
+    }
+
+    pub fn bootdisk_slots(&self) -> [i64; 2] {
+        match self {
+            Self::Gimlet | Self::Cosmo => [0x11, 0x12],
+        }
+    }
```

Comment:
this is much nicer than hard-coding it and the use-site, good call!

Reply:
Thanks!